[IMP] inventory: update two step doc copy (#9132)
odoo/documentation: content/applications/inventory_and_mrp/inventory/shipping_receiving/daily_operations/receipts_delivery_two_steps.rst

.. _inventory/receipts_delivery_two_steps:

.. |PO| replace:: :abbr:`PO (Purchase Order)`
.. |SO| replace:: :abbr:`SO (Sales Order)`
.. |DO| replace:: :abbr:`DO (Delivery Order)`
.. |RfQ| replace:: :abbr:`RfQ (Request for Quotation)`

Depending on a company's needs, receiving and shipping products in and out of the warehouse might
require multi-step operations. In Odoo *Inventory*, this can be done using *Multi-Step Routes*.

In the two-step receipt process, products are received in an input area, then transferred to stock.
This kind of process for incoming shipments might be beneficial for warehouses with specific storage
locations, such as freezers and refrigerators, secured locked areas, or special aisles and shelves.

Products can be sorted according to where they are going to be stored, and employees can stock all

> the products going to a specific location. The products are *not* available for further processing

Reviewer: ```suggestion
the products going to a specific location. The products are *not* available for further processing,
```

Author: Suggestion implemented ✅

   label at the end of the name.

Process receipt
---------------

From the warehouse receipt form, the products ordered can be received into the warehouse. To receive
the products, click :guilabel:`Validate`. Once validated, the receipt moves to the :guilabel:`Done`
stage, and the products move to the :guilabel:`WH/Input` location.

.. image:: receipts_delivery_two_steps/receipts-delivery-two-steps-receipt-form.png
   :align: center
   :alt: Receipt form for products ordered from vendor.

Click back to the |PO| (via the breadcrumbs, at the top of the form) to view the |PO| form. On the
product line, the quantity in the :guilabel:`Received` column now matches the ordered
:guilabel:`Quantity`.

Process internal transfer
-------------------------

Once the receipt is validated, an internal transfer is created and ready to process.

To view the internal transfer, navigate to the :menuselection:`Inventory app`, and locate the
:guilabel:`Internal Transfers` task card.

Click the :guilabel:`# To Process` button on the task card to reveal a list of all internal
transfers to process, and select the transfer associated with the previously validated receipt.

> Once ready, click :guilabel:`Validate` to complete the transfer and move the product from

Reviewer: ```suggestion
Once ready, click :guilabel:`Validate` to complete the transfer, and move the product from
```

Author: Suggestion implemented ✅

:guilabel:`WH/Input` to :guilabel:`WH/Stock`.

Once the transfer is validated, the products enter inventory, and are available for customer
deliveries or manufacturing orders.

.. image:: receipts_delivery_two_steps/receipts-delivery-two-steps-internal-transfer.png
   :align: center
   :alt: Internal transfer form for products ordered from vendor.

Process delivery order in two steps (pick + ship)
=================================================

When products are delivered in two steps, they move from warehouse stock to an output location.
Then, they move from the output location to a customer location in the database, upon validation of
a picking order, and a subsequent delivery order (DO).

Create sales order
------------------

To create a |SO|, navigate to the :menuselection:`Sales app`, and click :guilabel:`New`. This
opens a blank sales quotation form.

Add a customer in the :guilabel:`Customer` field. Then, fill out the various fields on the sales
quotation form, as necessary.

.. image:: receipts_delivery_two_steps/receipts-delivery-two-steps-new-sales-order.png
   :align: center
   :alt: Filled out new sales order form.

> Under the :guilabel:`Product` tab, click :guilabel:`Add a product`, and select a product to add to

Reviewer: ```suggestion
Under the :guilabel:`Order Lines` tab, click :guilabel:`Add a product`, and select a product to add to
```

Author: Suggestion implemented ✅

the sales order quotation.

Once ready, click :guilabel:`Confirm`. This moves the quotation to the :guilabel:`Sales Order`
stage.

Once the |SO| is confirmed, a :guilabel:`Delivery` smart button appears at the top of the form.
Clicking the smart button opens the warehouse delivery (WH/OUT) form.

.. image:: receipts_delivery_two_steps/receipts-delivery-two-steps-delivery-button.png
   :align: center
   :alt: Delivery smart button on validated sales order form.

Process picking
---------------

Once the sales order is confirmed, a picking order is generated and ready to process.

To complete the picking, navigate to the :guilabel:`Inventory app`, and locate the :guilabel:`Pick`
task card on the :guilabel:`Inventory Overview` dashboard. Alternatively, the picking order can also

> be accessed via the :guilabel:`Delivery` smart button at the top of the form.

Reviewer: ```suggestion
be accessed via the :guilabel:`Delivery` smart button at the top of the sales order form.
```

Author: Suggestion implemented ✅

> From the :guilabel:`Inventory Overview` page, click the :guilabel:`# To Process` button on the task

Reviewer: ```suggestion
From the :guilabel:`Inventory Overview` page, click the :guilabel:`# To Process` button on the :guilabel:`Pick` task
```

card. This reveals a list of all pickings to process.

Click on the picking (WH/PICK) operation associated with the sales order to reveal the picking
order.

.. image:: receipts_delivery_two_steps/receipts-delivery-two-steps-picking-form.png
   :align: center
   :alt: Picking order form for products included in sales order.

Click :guilabel:`Check Availability` on the picking order form. If the product is in stock, Odoo
automatically reserves the product. Quantities can also be manually set by changing the value in the
:guilabel:`Quantity` column to match the value in the :guilabel:`Demand` column.

Once ready, click :guilabel:`Validate` to complete the picking, and move the product from
:guilabel:`WH/Stock` to :guilabel:`WH/Output.`

Process delivery
----------------

> Once the picking is validated, a delivery order is created and ready to process.

Reviewer: ```suggestion
Once the picking is validated, a delivery order is created, and ready to process.
```

Author: Suggestion implemented ✅

.. image:: receipts_delivery_two_steps/receipts-delivery-two-steps-shipments.png
   :align: center
   :alt: Incoming and outgoing shipments set to two-step on warehouse form.

.. note::
   Selecting two-step receipts and deliveries automatically creates new *Input* and *Output*
   warehouse locations in the database, named `WH/Input` and `WH/Output`, respectively.

   To rename or edit these locations, navigate to :menuselection:`Inventory app --> Configuration
   --> Locations`, and select the desired location.

   On the location's form, change the :guilabel:`Location Name`, and make any other necessary
   changes.

Process receipt in two steps (input + stock)
============================================

When products are received in two steps, they move from the vendor location to an input location,
then from the input location to warehouse stock in the database upon validation of a purchase order
(PO) and a subsequent internal transfer.

Create purchase order
---------------------

To create a |PO|, navigate to the :menuselection:`Purchase app`, and click :guilabel:`New`. This
opens a blank :guilabel:`Request for Quotation` (RfQ) form.

> Add a vendor in the :guilabel:`Supplier` field. Then, fill out the various fields on the |RfQ|, as

Author: In 17.2 the field is changed from Vendor to "Supplier" in this instance

> Add a vendor in the :guilabel:`Supplier` field. Then, fill out the various fields on the |RfQ|, as

Reviewer: I, too, am only seeing a 'Vendor' field. It also seems to appear that way in 17.2 -- at least in the 17.2 runbot I used to double-check this. It might be worth looking into the accuracy of this field name during your revisions.

![Screenshot 2024-05-06 100913](https://github.com/odoo/documentation/assets/73958186/e90c16bf-7db4-4ad8-9a5a-425e10a6458b)

> Add a vendor in the :guilabel:`Supplier` field. Then, fill out the various fields on the |RfQ|, as

Author: I now am also seeing this. Not sure what happened to a few of the field names I saw upon first writing this... making the according changes now

necessary.

.. image:: receipts_delivery_two_steps/receipts-delivery-two-steps-new-rfq.png
   :align: center
   :alt: Filled out new Request for Quotation from vendor.

Under the :guilabel:`Products` tab, click :guilabel:`Add a product`, and select a product to add to
the |RfQ|.

Once ready, click :guilabel:`Confirm Order`. This moves the |RfQ| to the :guilabel:`Purchase Order`
stage.

Once the |PO| is confirmed, a :guilabel:`Receipt` smart button appears at the top of the form.
Clicking the smart button opens the warehouse receipt (WH/IN) form.

.. image:: receipts_delivery_two_steps/receipts-delivery-two-steps-smart-button.png
   :align: center

>    :alt: Delivery smart button for validated purchase order.

Reviewer 2: Why does it only say 1 receipt, and we have two steps enabled? may be good to say that the internal transfer is created later

Author: In 17.2 this is how it appears and works as intended per the change in routes functions

.. tip::
   For businesses with multiple warehouses that have different step configurations, the
   :guilabel:`Deliver To` field on the |PO| form might need to be specified as the correct *input

>    location* connected to the two-step warehouse.

Author: That's a good catch - it actually needs to be, not might. Changed wording here to be certain. Suggestion implemented ✅

product line, the quantity in the :guilabel:`Received` column now matches the ordered
:guilabel:`Quantity`.

Process internal transfer
-------------------------

Once the receipt is validated, an internal transfer is created and ready to process.

To view the internal transfer, navigate to the :menuselection:`Inventory app`, and locate the
:guilabel:`Internal Transfers` task card.

Click the :guilabel:`# To Process` button on the task card to reveal a list of all internal
transfers to process, and select the transfer associated with the previously validated receipt.

Once ready, click :guilabel:`Validate` to complete the transfer and move the product from
:guilabel:`WH/Input` to :guilabel:`WH/Stock`.

Once the transfer is validated, the products enter inventory, and are available for customer
deliveries or manufacturing orders.

.. image:: receipts_delivery_two_steps/receipts-delivery-two-steps-internal-transfer.png
   :align: center
   :alt: Internal transfer form for products ordered from vendor.

Process delivery order in two steps (pick + ship)
=================================================

> When products are delivered in two steps, they move from warehouse stock to an output location, then
> from the output location to a customer location in the database upon validation of a picking order,
> and a subsequent delivery order (DO).

Reviewer 2: ```suggestion
When products are delivered in two steps, they move from warehouse stock to an output location validation of a picking order. Then
they move from the output location to a customer location upon validating the delivery order (DO).
```

I'm not sure if this is functionally correct, but the original sentence seemed a bit dense, trying to simplify it a bit more.

Author: I reworded it a bit from your suggestion but nice call - I broke the sentence up a bit to avoid running on. Suggestion implemented ✅

Create sales order
------------------

To create a |SO|, navigate to the :menuselection:`Sales app`, and click :guilabel:`New`. This
opens a blank sales quotation form.

Add a customer in the :guilabel:`Customer` field. Then, fill out the various fields on the sales
quotation form, as necessary.

.. image:: receipts_delivery_two_steps/receipts-delivery-two-steps-new-sales-order.png
   :align: center
   :alt: Filled out new sales order form.

Under the :guilabel:`Product` tab, click :guilabel:`Add a product`, and select a product to add to
the sales order quotation.

Once ready, click :guilabel:`Confirm`. This moves the quotation to the :guilabel:`Sales Order`
stage.

> Once the |SO| is confirmed, a :guilabel:`Delivery` smart button appears at the top of the form.
> Clicking the smart button opens the warehouse delivery (WH/OUT) form.

Reviewer 2: ```suggestion
Once the |SO| is confirmed, a :guilabel:`Delivery` smart button appears at the top of the form.
Clicking the smart button opens the warehouse delivery (WH/OUT) form.
```
I'm seeing two delivery actions in runbot, your image shows one, why might this be?

![2024-05-03_10-10](https://github.com/odoo/documentation/assets/91896312/a062d556-b516-4861-b137-bfb778d24abe)

Author: In 17.2 the triggers for subsequent operations are different - populate with 1 instead of 2 and located elsewhere in the app

.. image:: receipts_delivery_two_steps/receipts-delivery-two-steps-delivery-button.png
   :align: center
   :alt: Delivery smart button on validated sales order form.

Process picking
---------------

Once the sales order is confirmed, a picking order is generated and ready to process.

> To complete the picking, navigate to the :guilabel:`Inventory app`, and locate the :guilabel:`Pick`
> task card on the :guilabel:`Inventory Overview` dashboard.

Reviewer 2: ```suggestion
To complete the picking, navigate to the :guilabel:`Inventory app`, and locate the :guilabel:`Pick`
task card on the :guilabel:`Inventory Overview` dashboard.
```

Alternatively, can I click the smart button?

Author: Yes! Added that in for clarification. Suggestion implemented ✅

Click the :guilabel:`# To Process` button on the task card. This reveals a list of all pickings to
process.

Click on the picking (WH/PICK) operation associated with the sales order to reveal the picking
order.

.. image:: receipts_delivery_two_steps/receipts-delivery-two-steps-picking-form.png
   :align: center
   :alt: Picking order form for products included in sales order.

> Click :guilabel:`Check Availability` on the picking order form. If the product is in stock, Odoo

Reviewer 2: ```suggestion
Click :guilabel:`Check Availability` on the picking order form. If the product is in stock, Odoo
```

I'm not showing this on the pick operation

Author: In 17.2 this shows on the form

Reviewer: Yeah, I'm in agreement with Tim on this once again. I even double-checked on a 17.2 runbot, and still wasn't seeing a 'Check Availability' option on the picking order form.
![Screenshot 2024-05-06 103254](https://github.com/odoo/documentation/assets/73958186/5a0b29ff-af39-47aa-9db9-9dc4487e61b0)

Author: Turns out the check availability button was showing in Master, not 17.2... so I'm making the necessary changes here

automatically reserves the product. Quantities can also be manually set by changing the value in the
:guilabel:`Quantity` column to match the value in the :guilabel:`Demand` column.

Once ready, click :guilabel:`Validate` to complete the picking, and move the product from
:guilabel:`WH/Stock` to :guilabel:`WH/Output.`

Process delivery
----------------

> Once the picking is validated, a delivery order is created and ready to process.

Reviewer 2: ```suggestion
Once the picking is validated, a delivery order is created and ready to process.
```

This was created for me already when confirming the sales order

Author: In 17.2 it's not created until after validation of the previous order

Reviewer: This is another instance where I think you'll need to double-check things to ensure accuracy, because, even when I check things on a 17.2 runbot, Tim and I are getting the same results.

Author: What is generated upon confirmation of an SO is not a delivery order, it's the delivery smart button - which if clicked contains the picking order. Only when the picking order is validated does an actual delivery order generate - which is also contained in the delivery smart button. I'll reword this section to clarify this, but it is functionally correct as is

To view the delivery order, navigate back to the :guilabel:`Inventory Overview` page, via the
breadcrumbs, and locate the :guilabel:`Delivery Orders` task card.

Click the :guilabel:`# To Process` button on the task card to reveal a list of all delivery orders
to process, and select the order associated with the previously validated picking.

.. image:: receipts_delivery_two_steps/receipts-delivery-two-steps-delivery-order.png
   :align: center
   :alt: Delivery order form for products ordered by customer.

> To deliver the products, change the value in the :guilabel:`Quantity` field to match the ordered
> quantity in the :guilabel:`Demand` field.

Reviewer 2: ```suggestion
To deliver the products, change the value in the :guilabel:`Quantity` field to match the ordered
quantity in the :guilabel:`Demand` field.
```

This was done automatically for me.

Author: In 17.2 it's not done automatically, and only after completion of the previous steps

Reviewer: Tim and I are getting the same result here, as well

Once ready, click :guilabel:`Validate`. Once validated, the delivery order moves to the
:guilabel:`Done` stage.

> Click back to the |SO| (via the breadcrumbs, at the top of the form) to view the |SO| form. On the

Reviewer 2: ```suggestion
Click back to the |SO| (via the breadcrumbs, at the top of the form) to view the |SO| form. On the
```

I couldn't click back to the SO via breadcrumbs because I left the sales app to go into the inventory app on line 186

Author: Nice catch - changing this accordingly. Suggestion implemented ✅
